analysis/table1.R

```diff
@@ -0,0 +1,216 @@
+  "_clean.rds"
+)) # "_stage1.rds"))
+
+# Table 1 Processing Start -----------------------------------------------------
+print("Table 1 processing")
+
+# Remove people with history of COVID-19 ---------------------------------------
+print("Remove people with history of COVID-19")
+
+df <- df[df$sub_bin_covidhistory == FALSE, ]
+
+# Create exposure indicator ----------------------------------------------------
+print("Create exposure indicator")
+
+df$exposed <- !is.na(df$exp_date_covid)
+
+# Define age groups ------------------------------------------------------------
+print("Define age groups")
+
+df$cov_cat_age_group <- numerical_to_categorical(df$cov_num_age, age_bounds) # See utility.R
+df$cov_cat_age_group <- ifelse(
+  df$cov_cat_age_group == "<=17",
+  "",
+  df$cov_cat_age_group
+) # for consistency in blanking out underage
+
+df$cov_cat_consrate2019 <- numerical_to_categorical(
+  df$cov_num_consrate2019,
+  c(1, 6),
+  zero_flag = TRUE
+)
+
+median_iqr_string <- paste0(
+  quantile(df$cov_num_age)[3],
+  " (",
+  quantile(df$cov_num_age)[2],
+  "-",
+  quantile(df$cov_num_age)[4],
+  ")"
+)
+
+# Filter data ------------------------------------------------------------------
+print("Filter data")
+
+df <- df[, c(
+  "patient_id",
+  "exposed",
+  colnames(df)[grepl("cov_cat_", colnames(df))],
+  colnames(df)[grepl("strat_cat_", colnames(df))],
+  colnames(df)[grepl("cov_bin_", colnames(df))]
+)]
+
+df$All <- "All"
+
+# Filter binary data
+
+for (colname in colnames(df)[grepl("cov_bin_", colnames(df))]) {
+  df[[colname]] <- sapply(df[[colname]], as.character)
+}
+
+df <- df %>%
+  mutate(across(where(is.factor), as.character))
+
+# Aggregate data ---------------------------------------------------------------
+print("Aggregate data")
+
+df <- tidyr::pivot_longer(
+  df,
+  cols = setdiff(colnames(df), c("patient_id", "exposed")),
+  names_to = "characteristic",
+  values_to = "subcharacteristic"
+)
+#as.factor or factor to convert binary functions
+
+df$total <- 1
+
+df <- aggregate(
+  cbind(total, exposed) ~ characteristic + subcharacteristic,
+  data = df,
+  sum
+)
+
+# Tidy care home characteristic ------------------------------------------------
+print("Remove extraneous information")
+
+df <- df[df$subcharacteristic != FALSE, ]
+df$subcharacteristic <- ifelse(
+  df$subcharacteristic == "" | df$subcharacteristic == "unknown",
+  "Missing",
+  df$subcharacteristic
+)
+
+# Sort characteristics ---------------------------------------------------------
+print("Sort characteristics")
+
+df <- df[order(df$subcharacteristic, decreasing = TRUE), ] # Basic sort from extendedtable, might need to clean up
+df <- df[order(df$characteristic), ]
+
+# Add in Median IQR
+print('Add median (IQR) age')
+
+# Pastes: "Mean Age (LQ Age - UQ Age)" as a string for each cohort
+df[nrow(df) + 1, ] <- c("Age, years", "Median (IQR)", median_iqr_string, 0)
+
+# Define table1 output folder ---------------------------------------------------------
```

Comment:
We define the output folder near the start of `dataset_clean.R`. We should agree a consistent approach to use across the repo.